csharp/src/Ice/BinaryConnection.cs

```diff
@@ -0,0 +1,158 @@
+//
+// Copyright (c) ZeroC, Inc. All rights reserved.
+//
+
+using System;
+using System.Diagnostics;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+using ZeroC.Ice.Instrumentation;
+
+namespace ZeroC.Ice
+{
+    public abstract class BinaryConnection : IAsyncDisposable
```

Comment:
Why is this public, seems the only public method is DisposeAsync from IAsyncDisponsable

Reply:
Yes, sorry I forgot to add a TODO here... this is needed because of the XxxConnection constructors which are public are passing around an BinaryConnection parameter. We need to come up with a better plan for the pluggable transport API (and decide what should be public and what shouldn't). I'll put some thoughts into it but not for this PR 😄.